code/modules/reagents/chemistry/reagents/impure_reagents/impure_medicine_reagents.dm

```diff
@@ -540,3 +544,207 @@
 	REMOVE_TRAIT(owner, TRAIT_NODEATH, type)
 	owner.remove_movespeed_modifier(/datum/movespeed_modifier/reagent/nooartrium)
 	owner.remove_actionspeed_modifier(/datum/actionspeed_modifier/nooartrium)
+
+/*				Non c2 medicines 				*/
+
+/datum/reagent/impurity/mannitol
+	name = "Mannitoil"
+	description = "Gives the patient a temporary speech impediment."
+	color = "#CDCDFF"
+	addiction_types = list(/datum/addiction/medicine = 5)
+	ph = 12.4
+	liver_damage = 0
+	///The speech we're forcing on the owner
+	var/speech_option
+
+/datum/reagent/impurity/mannitol/on_mob_add(mob/living/owner, amount)
+	. = ..()
+	if(!iscarbon(owner))
+		return
+	var/mob/living/carbon/carbon = owner
+	speech_option = pick(list(SWEDISH, UNINTELLIGIBLE, STONER, MEDIEVAL, WACKY, NERVOUS, MUT_MUTE))
+	carbon.dna?.activate_mutation(speech_option)
```

Comment:
what if that speech option isn't in their DNA?

Reply:
Mutations are species specific? I had no idea, I'll try to read into the code about it.

Reply:
I can't find anything that shows that it would be species specific, but I did improve it so that it won't add one that's already there.